fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountInterestPostingServiceImpl.java

```diff
@@ -268,16 +271,66 @@ public List<PostingPeriod> calculateInterestUsing(final MathContext mc, final Lo
             if (postedAsOnDates.contains(periodInterval.endDate().plusDays(1))) {
                 isUserPosting = true;
             }
-            final PostingPeriod postingPeriod = PostingPeriod.createFromDTO(periodInterval, periodStartingBalance,
-                    retreiveOrderedNonInterestPostingTransactions(savingsAccountData), monetaryCurrency, compoundingPeriodType,
-                    interestCalculationType, interestRateAsFraction, daysInYearType.getValue(), upToInterestCalculationDate,
-                    interestPostTransactions, isInterestTransfer, minBalanceForInterestCalculation,
-                    isSavingsInterestPostingAtCurrentPeriodEnd, overdraftInterestRateAsFraction, minOverdraftForInterestCalculation,
-                    isUserPosting, financialYearBeginningMonth, savingsAccountData.isAllowOverdraft());
+            if (savingsAccountData.isAllowOverdraft() && !MathUtil.isZero(savingsAccountData.getGlAccountIdForInterestReceivable())) {
+
+                List<SavingsAccountTransactionData> overdraftTransactionsInPeriod = listForOverdraft(savingsAccountData, periodInterval);
+                List<SavingsAccountTransactionData> interestPostingTransactionsInPeriod = listForInterestPosting(savingsAccountData,
+                        periodInterval);
+                Boolean isOverdraftAccountType = isOverdraftAccount(savingsAccountData, periodInterval);
+                List<SavingsAccountTransactionData> prioritizedTransactionList = null;
+                List<SavingsAccountTransactionData> fallbackTransactionList = null;
 
-            periodStartingBalance = postingPeriod.closingBalance();
+                Boolean flagIntroduce = false;
```

Comment:
`flagIntroduce` what is this for? I dont like the naming. Please give a more descriptive name to this field!